src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -157,6 +201,47 @@ DecisionTree<FitnessFunction,
       minimumLeafSize, minimumGainSplit);
 }
 
+//! Construct and train with weights.
+template<typename FitnessFunction,
+        template<typename> class NumericSplitType,
+        template<typename> class CategoricalSplitType,
+        typename DimensionSelectionType,
+        typename ElemType,
+        bool NoRecursion>
+template<typename MatType, typename LabelsType, typename WeightsType>
+DecisionTree<FitnessFunction,
+        NumericSplitType,
+        CategoricalSplitType,
+        DimensionSelectionType,
+        ElemType,
+        NoRecursion>::DecisionTree(const DecisionTree& other,
+                                   MatType data,
+                                   LabelsType labels,
+                                   const size_t numClasses,
+                                   WeightsType weights,
+                                   const size_t minimumLeafSize,
+                                   const double minimumGainSplit,
+                                   const std::enable_if_t<
+                                           arma::is_arma_type<
+                                                   typename std::remove_reference<
+                                                           WeightsType>::type>::value>*):
```

Comment:
Probably this could look a little cleaner if you just wrap the first parameter `const DecisionTree& other` to four characters indented on a new line and have the rest match that.